### ISSUE TITLE
Power from RPI - Not charging

### ISSUE THREAD

**Issue by Ov1** [PiSupply/PiJuice#89]
Pi3, PiJuice upgraded to firmware 1.1

I have noted the following issues when **power is being fed to the RPI directly**:

- PiJuice tray icon does not show up.
- Battery does not charge.
- PiJuice doesn't know when power is switched off or on.

I have checked the pijuice service and it is running. I even tried to stop and start the service but this does not change anything. If I reboot the Pi, the battery starts charging and the battery status also changes to "Normal" after the first power loss. However, any subsequent change in power does not affect the battery status and it does not charge again. Only after a reboot again will the battery start charging, but then again it will stop charging after power is switched off and back on.

This effectively makes the PiJuice useless if power is fed to the RPI directly.
However, if power is fed to the PiJuice directly, everything works.

**Comment by tvoverbeek** [comment]
There is a bug in pijuice_tray.py which makes the tray app crash when power is not supplied via the PiJuice usb (= crash when power from the RPi or power from battery only).
To fix it see the following diff between the corrected version and the version in /usr/bin/pijuice_tray.py:
```
--- pijuice_tray.py	2018-04-01 22:05:53.149123763 +0200
+++ /usr/bin/pijuice_tray.py	2018-03-19 21:02:51.000000000 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python
 # -*- coding: utf-8 -*-
 from __future__ import print_function, division
 
@@ -154,9 +154,9 @@
                 elif self.status['battery'] == 'CHARGING_FROM_IN' or self.status['powerInput'] != 'NOT_PRESENT':
                     b_file = ICON_DIR + '/bat-in-' + str((b_level//10)*10) + '.png'
                 elif self.status['battery'] == 'CHARGING_FROM_5V_IO' or self.status['powerInput5vIo'] != 'NOT_PRESENT':
-                    b_file = ICON_DIR + '/bat-rpi-' + str((b_level//10)*10) + '.png'
+                    b_file = ICON_DIR + '/bat-rpi-' + str((b_level/10)*10) + '.png'
                 else:
-                    b_file = ICON_DIR + '/bat-' + str((b_level//10)*10) + '.png'
+                    b_file = ICON_DIR + '/bat-' + str((b_level/10)*10) + '.png'
             else:
                 b_file = ICON_DIR + '/connection-error.png'
                 if py3 == False:
```
The corrected version is the one prefixed with `---`

Is charging enabled on the General tab of the PiJuice HAT Configuration? 
I can remove power from the Pi and when I reconnect it PiJuice continues charging.

**Comment by Ov1** [comment]
Thanks for your reply. I have corrected /usr/bin/pijuice_tray.py as per your provided diff. The tray icon now shows when power is from the RPi. But PiJuice still does not charge the battery when power is cut and restored later.

Charging is already enabled on the HAT configuration. I'm including screenshots for your information. Power is provided from the RPi USB.

PiJuice is charging correctly:
![2018-04-10-180438_1824x984_scrot](https://user-images.githubusercontent.com/4319777/38562623-13af7cda-3cec-11e8-9ab3-7aa7b4019bd3.png)

Now, power is cut:
![2018-04-10-180457_1824x984_scrot](https://user-images.githubusercontent.com/4319777/38562781-68c0b2d4-3cec-11e8-8934-d2c487afab2f.png)

Power is then restored:
![2018-04-10-180523_1824x984_scrot](https://user-images.githubusercontent.com/4319777/38562850-916eb262-3cec-11e8-973b-eebcac3b4f5b.png)

You'll notice that in the last screenshot, the battery is not being charged (PiJuice leds confirm this). If I leave everything as it is and reboot the Pi, the battery will start charging as shown by the first screenshot.

Any ideas?

**Comment by tvoverbeek** [comment]
Maybe your power supply does not provide enough power (or the voltage is too low).
On your first screenshot it says the GPIO power input is 'WEAK'.
Here is a screenshot from my PiJuice while charging from GPIO:
<img width="502" alt="screen shot 2018-04-10 at 18 44 09" src="https://user-images.githubusercontent.com/1197294/38571075-c9b79190-3cef-11e8-8c9c-c324cc66748d.png">
Note in my case GPIO power input has 5.108V and it says 'PRESENT' instead of 'WEAK'.
Can you try a different power supply (possibly with higher capacity)?
Note, the Pi 3B uses more power than previous models and the Pi 3B+ even more.

**Comment by Ov1** [comment]
You were right. The problem seems to be related to the power adapter. I tried the following adapters:

- 5.0V / 0.7A. Samsung. Works fine.
- 5.2V / 1.35A. ASUS. Works fine.
- 5.1V / 2.5A. RPi Official Adapter. Works fine. 

The adapter with the problem was the one provided by PiSupply as part of the KickStarter pledge. It's a 5.25V / 1.5A. I got 2 and both have these issues. You'll notice that charging works fine even with the 0.7A one. So I guess something is wrong with the power adapter itself.

Thank you for taking the time in troubleshooting this. Much appreciated.
